rust/agama-dbus-server/src/network/model.rs

```diff
@@ -63,7 +65,21 @@ impl NetworkState {
             return Err(NetworkStateError::ConnectionExists(conn.uuid()));
         }
 
+        // FIXME: unify this code.
+        match &conn {
+            Connection::Bond(bond) => {
+                let id = conn.id().to_string();
+                // FIXME: We should implement a trait Controller
+                // so those that acts as a controller offer the same API
+                // and we do not need to access "bond.bond".
+                let ports = bond.bond.ports.clone();
+                self.update_controller_ports(id, ports)?;
+            }
+            _ => {}
+        };
+
```

Comment:
What if someone update a Connection which is a port. Do we need to update the Controller as well (if it is Bond). Something like: 
```
if let Some(ctrl_id) = old_conn.controller() {
    if let Some(conn) = self.get_connection_mut(ctrl_id) {
        if let Connection::Bond(bond) = conn {
            let id = old_conn.id().to_string();
            if ! bond.bond.ports.contains(&id) {
                bond.bond.ports.push(id);
            }
        }
    }
}
```

Reply:
In general I think we should "declare", what need to be there first. The Bond connection or the Port connection. And depending on this, maybe we should throw an error if the corresponding reference doesn't exists.

I would prefere not having `bond.bond.ports` but replacing it with a method which creates a iterator on the fly with all corresponding ports.

Reply:
Sure! Please have in mind that the PR is unfinished and not ready for review yet, although early comments are appreciated. There are many things, especially error handling, that we need to implement/improve. And, of course, we will replace the `bond.bond` thingie with something better as it hurts my eyes too :-)

Reply:
> What if someone update a Connection which is a port. Do we need to update the Controller as well (if it is Bond). Something like:

Good point, we need to handle that case too. It is not possible with the D-Bus API (you always set the ports through the Bond), but it is something that we need to tackle.

Reply:
We decided to simplify things and only keep the reference in the port. And, btw, we are using the UUID so if we rename any interface/connection we do not need to update the links.